samples: samples/subsys/tracing is broken for UART tracing

**Describe the bug**
Follow the [ReadMe](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/tracing/README.txt) to build the UART tracing sample:

```
cmake -B bld_trace -GNinja -DBOARD=mps2_an521 -S samples/subsys/tracing/ -DCONF_FILE=prj_uart.conf
cd bld_trace
ninja
```

The build failed since the **first** trackable commit of `samples/subsys/tracing/src/main.c` (eaba71738e1044eede2e53dd1336fe5098a987ea).

Error as below:

```
/home/xxx/zephyrproject2/zephyr/include/zephyr/device.h:96:39: error: '__device_dts_ord_DT_CHOSEN_zephyr_tracing_uart_ORD' undeclared (first use in this function)
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/home/xxx/zephyrproject2/zephyr/include/zephyr/device.h:276:37: note: in expansion of macro 'DEVICE_NAME_GET'
  276 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/xxx/zephyrproject2/zephyr/include/zephyr/device.h:296:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  296 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/xxx/zephyrproject2/zephyr/subsys/tracing/tracing_backend_uart.c:79:21: note: in expansion of macro 'DEVICE_DT_GET'
   79 |  tracing_uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_tracing_uart));
      |                     ^~~~~~~~~~~~~
/home/xxx/zephyrproject2/zephyr/include/zephyr/device.h:96:39: note: each undeclared identifier is reported only once for each function it appears in
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/home/xxx/zephyrproject2/zephyr/include/zephyr/device.h:276:37: note: in expansion of macro 'DEVICE_NAME_GET'
  276 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/xxx/zephyrproject2/zephyr/include/zephyr/device.h:296:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  296 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/xxx/zephyrproject2/zephyr/subsys/tracing/tracing_backend_uart.c:79:21: note: in expansion of macro 'DEVICE_DT_GET'
```

--- a/samples/subsys/tracing/README.txt
+++ b/samples/subsys/tracing/README.txt
@@ -18,6 +18,9 @@ or:
 
     cmake -DBOARD=mps2_an521 -DCONF_FILE=prj_uart_ctf.conf ..
 
+NOTE: You may need to set 'zephyr,tracing-uart' property under the chosen
+node in your devicetree.  See boards/mps2_an521.overlay for an example.
+
 After the application has run for a while, check the trace output file.
 
 --------------------------------------------------------------------------------

--- a/samples/subsys/tracing/boards/mps2_an521.overlay
+++ b/samples/subsys/tracing/boards/mps2_an521.overlay
@@ -1,0 +1,7 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/ {
+       chosen {
+               zephyr,tracing-uart = &uart1;
+       };
+};

--- a/samples/subsys/tracing/sample.yaml
+++ b/samples/subsys/tracing/sample.yaml
@@ -18,6 +18,7 @@ tests:
   tracing.transport.uart:
     platform_allow: qemu_x86 qemu_x86_64
     extra_args: CONF_FILE="prj_uart.conf"
+    filter: dt_chosen_enabled("zephyr,tracing-uart")
   tracing.transport.usb:
     platform_allow: sam_e70_xplained
     depends_on: usb_device
@@ -25,6 +26,7 @@ tests:
   tracing.transport.ctf:
     platform_allow: qemu_x86 qemu_x86_64
     extra_args: CONF_FILE="prj_uart_ctf.conf"
+    filter: dt_chosen_enabled("zephyr,tracing-uart")
   tracing.transport.ctf:
     platform_allow: sam_e70_xplained
     depends_on: usb_device

